litestar/response/redirect.py

```diff
@@ -94,6 +97,7 @@ def __init__(
         media_type: str | MediaType | None = None,
         status_code: RedirectStatusType | None = None,
         type_encoders: TypeEncodersMap | None = None,
+        query_params: Mapping[str, str | Sequence[str]] | None = None,
```

Comment:
This should support `datastructures.MultiDict` instances, as that's what we're returning from `URL.query_params`